pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotRealtimeSegmentManager.java

```diff
@@ -138,15 +138,22 @@ private synchronized void assignRealtimeSegmentsToServerInstancesIfNecessary()
         for (String instanceId : instancesInResource) {
           InstanceZKMetadata instanceZKMetadata = _pinotHelixResourceManager.getInstanceZKMetadata(instanceId);
           String groupId = instanceZKMetadata.getGroupId(resource);
+          String partitionId = instanceZKMetadata.getPartition(resource);
+          AbstractTableConfig realtimeTableConfig =
+              ZKMetadataProvider.getRealtimeTableConfig(_pinotHelixResourceManager.getPropertyStore(), resource);
+          if (instancesInResource.size() == Integer.parseInt(realtimeTableConfig.getValidationConfig().getReplication())) {
```

Comment:
We default using "ALL" and we always assign partitionId = 0 to each instance when there is only one partition in a replica.
I'm fine with assign only parititonId in segment name. Just the segment naming convention won't be backward compatible.
This won't change routing table build, so the broker logic is fine.